DesignDoc.md

```diff
@@ -0,0 +1,157 @@
+
+
+### 4.1. In Scope
+
+* looking at reviews from others people
+* search movies by title director and actor
+* show popular movies to leave reviews on.
+* Adding, updating, deleting, and viewing reviews.
+
+
+### 4.2. Out of Scope
+
+* following other users .
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+We will use API Gateway and Lambda to create endpoints that will handle the creation, updating, and retrieval of movies and reviews. 
+
+We will store movie review information in a table in DynamoDB.
+
+ReelDeal will also provide a web interface for users. A main page providing a variety of options to add, update and reminisce of all the movies you watched and maybe some you warn others not to watch.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package._
+
+```
+// Movie 
+    private int id;
+    private String title;
+    private String description;
+    private String releaseDate;
+    private String posterUrl;
+    private List<String> genres;
+    private List<String> cast;
+    private String director;
+
+
+// Review 
+    private int id;
+    private int movieId;
+    private String username;
+    private String text;
+    private int rating;
+    private Date date;
+
+
+// User 
+    private int id;
+    private String username;
+    private String password;
+    private String email;
+    private List<Review> reviews;
+
+
+
+```
+
+GET /movies - Retrieve a list of all movies
+GET /movies/{id} - Retrieve a specific movie by its ID
+POST /movies - Create a new movie
+PUT /movies/{id} - Update an existing movie
+DELETE /movies/{id} - Delete a movie
+
+GET /movies/{id}/reviews - Retrieve all reviews for a specific movie
+GET /reviews/{id} - Retrieve a specific review by its ID
+POST /reviews - Create a new review
+PUT /reviews/{id} - Update an existing review
+DELETE /reviews/{id} - Delete a review
+
+GET /users - Retrieve a list of all users
+GET /users/{id} - Retrieve a specific user by its ID
+GET /users/{username} - Retrieve a specific user by its username
+POST /users - Create a new user
+PUT /users/{id} - Update an existing user
+DELETE /users/{id} - Delete a user
+
+
+# 7. Tables
+
+_Define the DynamoDB tables you will need for the data your service will use. It may be helpful to first think of what objects your service will need, then translate that to a table structure, like with the *`Playlist` POJO* versus the `playlists` table in the Unit 3 project._
+
+Movies:
+- Table primary key: id (String)
```

Comment:
I think you mean partition key for this and the other tables? (instead of primary key)

Reply:
What are you thinking you'll use the ID field for? (I''m wondering if it's needed... for all of your tables)

Reply:
I always thought it was easiest to search for entries based on Id fields.